starter/source/interfaces/interf1/inintr.F

```diff
@@ -201,100 +210,158 @@ SUBROUTINE ININTR(IPARI     ,INSCR    ,X,V       ,IXS      ,IXQ      ,
-         IREMNODE = 1
-         NREMNODE = 16*IPARI(4,N)
-         IF(NREMNODE < 0)THEN
-           NREMNODE=1073741824  ! decrease initial value of NREMNODE to fit into integer storage
-         END IF
-         CALL UPGRADE_REMNODE(IPARI(1,N),NREMNODE,INTBUF_TAB(N),NTY)
-       ENDIF
-       IF (NTY == 11.AND. IPARI(63,N) == 2 .AND. IDDLEVEL == 1)THEN
+        ! ----------------
+        ! loop over the interfaces
+        ! interface type 24 must be treated at the end
+        ! interface type 2 must be treated before interface 24
+        ! other interfaces are treated at the beginning
+        NEXT_INTER(1:3) = 0
+        DO N=1,NINTER
+          NTY=IPARI(7,N)
+          IF(NTY==2) THEN
+            NEXT_INTER(2) = NEXT_INTER(2) + 1
+            INTERFACE_TYPE2(NEXT_INTER(2)) = N
+          ELSEIF(NTY==24) THEN
+            NEXT_INTER(3) = NEXT_INTER(3) + 1
+            INTERFACE_TYPE24(NEXT_INTER(3)) = N
+          ELSE
+            NEXT_INTER(1) = NEXT_INTER(1) + 1
+            INTERFACE_OTHER(NEXT_INTER(1)) = N
+          ENDIF
+        ENDDO
+        ADDRESS_INTER(1) = 0
+        ADDRESS_INTER(2) = NEXT_INTER(1)
+        ADDRESS_INTER(3) = NEXT_INTER(1) + NEXT_INTER(2)
+        INTERFACE_INDEX(1:NEXT_INTER(1)) = INTERFACE_OTHER(1:NEXT_INTER(1))
+        INTERFACE_INDEX(ADDRESS_INTER(2)+1:ADDRESS_INTER(2)+NEXT_INTER(2)) = INTERFACE_TYPE2(1:NEXT_INTER(2))
+        INTERFACE_INDEX(ADDRESS_INTER(3)+1:ADDRESS_INTER(3)+NEXT_INTER(3)) = INTERFACE_TYPE24(1:NEXT_INTER(3))
+        ! ----------------
+        IWRN = 0
+        NINT2 = 0
+        NREMN(1:NINTER) = 0     
+        ! ----------------
+        ! loop over the interfaces
+        ! 1) interface type 1,3:23,25 + other
+        !    forbidden S nodes (defined in remnode structure) will BE take into account in the domain decomposition
+        ! 2) interface type 2
+        !    at the end of the interface type 2 treatment, update :
+        !    a) remnode algo for interface type 2 
+        !    b) remnode algo for interface type 24
+        ! 3) interface type 24
+        !    forbidden S nodes (defined in remnode structure) will not be take into account in the domain decomposition
+        DO KIND_INTERFACE=1,3
+          
+          ! ----------------
+          DO IJK=1,NEXT_INTER(KIND_INTERFACE)
+            N = INTERFACE_INDEX(ADDRESS_INTER(KIND_INTERFACE)+IJK)
+            IREMNODE = 0
+            IREMNODE_EDG = 0
+            NREMNODE = 0
+            NTY=IPARI(7,N)
+            FLAG_REMOVED_NODE = .FALSE.
+            IF ((NTY == 7.OR.NTY == 25) .AND. IPARI(63,N) == 2 .AND. IDDLEVEL == 1)THEN
 C---     Initial dimension of REMNODE arrays
-         IREMNODE = 1
-         REMNODE_SIZE = 5*IPARI(4,N)
-         CALL UPGRADE_REMNODE(IPARI(1,N),REMNODE_SIZE,INTBUF_TAB(N),NTY)
-       ENDIF
-       IF (NTY == 25.AND.IPARI(58,N) >0 .AND. IPARI(63,N) == 2 .AND. IDDLEVEL == 1)THEN
+              IREMNODE = 1
+              NREMNODE = 16*IPARI(4,N)
+              IF(NREMNODE < 0)THEN
```

Comment:
Integer overflow cannot be detected looking at the sign of the result anymore. This behavior changed with GCC13
[see here](https://gcc.gnu.org/gcc-13/porting_to.html ).
Instead, you can test `IPARI(4,n) < HUGE(N) / 16` instead of the sign of NREMNOD